packages/js/src/utils/id-token.ts

```diff
@@ -0,0 +1,39 @@
+import { UrlSafeBase64 } from '@silverhand/essentials';
+import * as s from 'superstruct';
+
+/**
+ * @link [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)
+ */
+const IdTokenClaimsSchema = s.type({
+  iss: s.string(),
+  sub: s.string(),
+  aud: s.string(),
+  exp: s.number(),
+  iat: s.number(),
+  at_hash: s.optional(s.string()),
+});
+
+export type IdTokenClaims = s.Infer<typeof IdTokenClaimsSchema>;
+
+export const decodeIdToken = (token: string): IdTokenClaims => {
+  const { 1: encodedPayload } = token.split('.');
+
+  if (!encodedPayload) {
+    throw new Error('invalid token');
```

Comment:
Could refer to the Error definitions in Kotlin or Swift SDKs. Could define multiple type of errors for specific exceptions case. e.g. throw new JWTError('Invalid JWT format'). or throw new JWTError('Invalid token claims')

P.S Do we have a test case for this?

Reply:
Conclusion: Do not define custom Error for now, and do it when needed ( See [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1642588826359200)

Reply:
still need a custom error here. would you like to add a linear issue before merging?

Reply:
Added LOG-1355